src/divapi_js_minifier.erl

```diff
@@ -0,0 +1,147 @@
+    divapi_cache:get(
+        {diversity_json, ComponentName, Tag},
+        fun () -> jiffy:decode(git_utils:get_diversity_json(ComponentName, Tag)) end,
+        1000 * 60 * 60 * 5 % 5 hours
+     ).
+
+%% @doc Fetches a list of already minified tags for a specific component.
+get_minified_tags(ComponentName) ->
+    MinifiedPath = get_minify_path(ComponentName),
+    case file:list_dir(MinifiedPath) of
+        {ok, TagList} -> lists:map(fun(Tag) -> unicode:characters_to_binary(Tag) end, TagList);
+        {error, _}    -> []
+    end.
+
+%% @doc Returns path to where each minified file is stored for component.
+get_minify_path(ComponentName) ->
+    MinifiedPath = case application:get_env(divapi, component_cache) of
+        undefined  -> <<"/var/diversity-api/js_minified">>;
+        {ok, Dir}  -> <<(unicode:characters_to_binary(Dir))/binary, "/js_minified">>
+    end,
+    filename:join(MinifiedPath, <<ComponentName/binary, "/">>).
+
+get_file_type(<<"//", _Rest/binary>> = ScriptFile) ->
+    %% This is an exteral js file that needs to be downloaded.
+    {external, <<"http:", ScriptFile/binary>>};
+get_file_type(<<"http://", _Rest/binary>> = ScriptFile) ->
+    %% This is an exteral js file that needs to be downloaded.
+    {external, ScriptFile};
+get_file_type(<<"https://", _Rest/binary>> = ScriptFile) ->
+    %% This is an exteral js file that needs to be downloaded.
+    {external, ScriptFile};
+get_file_type(ScriptFile) ->
+    case re:run(ScriptFile, <<".min.js">>) of
+        {match, _} ->
+            %% It's already minified. And exists in our component already
+            {already_minified, ScriptFile};
+        _ ->
+            %% It's not a minified file. Assume internal
+            {jsfile, ScriptFile}
+    end.
+
+%% Fetch filec content from outside. And add it to the filecontent
+get_file_contents(ComponentName, Tag, [{external, ScriptFilePath} | Rest], FileContent) ->
+    Opts = [{body_format, binary}],
+    Request = {binary_to_list(ScriptFilePath), []},
+    ScriptFile = case httpc:request(get, Request, [], Opts) of
+        {ok, {{_Version, Status, _ReasonPhrase}, _Headers, Body}} ->
+
+            case Status of
+                404 -> throw({resource_not_found, ScriptFilePath});
+                500 -> throw({server_error, ScriptFilePath});
+                200 -> Body
+            end
+    end,
+    get_file_contents(ComponentName, Tag, Rest, <<FileContent/binary, ";", ScriptFile/binary>>);
+%% No action needed, add it to the file acc as is
+get_file_contents(ComponentName, Tag, [{already_minified, ScriptFilePath} | Rest], FileContent) ->
+    FileBody = git_utils:get_file(ComponentName, Tag, ScriptFilePath),
+    get_file_contents(ComponentName, Tag, Rest, <<FileContent/binary, ";", FileBody/binary>>);
+%% Fetches filecontent for component and minify it through uglify
+get_file_contents(ComponentName, Tag, [{jsfile, ScriptFilePath} | Rest], FileContent) ->
+    TmpName0 = {ComponentName, Tag, ScriptFilePath},
```

Comment:
Detta är inte nödvändigt för javascript då datan är statisk. Låt de ha kvar sina originalnamn med .min.js som ändelse eller något. Detta gjordes i SASS-fallet för att vi berodde på variablerna in.